tests/DataTypes/TimezoneTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/*
+ * This file contains the Tests\CBS\SmarterU\TimezoneTest.
+ *
+ * @author Brian Reich <brian.reich@thecoresolution.com>
+ * @copyright $year$ Core Business Solutions
+ * @license Proprietary
+ * @since 2023/03/27
+ * @version $version$
+ */
+
+declare(strict_types=1);
+
+namespace Tests\CBS\SmarterU\DataTypes;
+
+use CBS\SmarterU\DataTypes\Timezone;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests Timezone
+ */
+class TimezoneTest extends TestCase {
+
+    #region Tests that are possibly too much
+
+    /**
+     * Verifies that getProvidedNameFromDisplayValue() returns the right value
+     * when a valid display value is passed.
+     *
+     * @dataProvider displayValueProvider
+     */
+    public function testGetProvidedNameFromDisplayValueReturnsMatchingProvidedNameForValidDisplayValue(string $displayValue) {
+        $flippedTimezones = array_flip(Timezone::VALID_TIMEZONES_PROVIDED_NAME_TO_DISPLAY_VALUE);
+
+        $this->assertEquals(
+            $flippedTimezones[$displayValue],
+            Timezone::getProvidedNameFromDisplayValue($displayValue)
+        );
+    }
+
+    /**
+     * Verifies that getDisplayValueFromProvidedName returns the right value
+     * when a valid provided name is passed.
+     *
+     * @dataProvider providedNameProvider
+     */
+    public function testGetDisplayValueFromProvidedNameReturnsMatchingDisplayValueForValidProvidedName(string $providedName) {
+        $this->assertEquals(
+            Timezone::VALID_TIMEZONES_PROVIDED_NAME_TO_DISPLAY_VALUE[$providedName],
+            Timezone::getDisplayValueFromProvidedName($providedName)
+        );
+    }
```

Comment:
I am going to agree that this is probably too much... I think you are simply verifying that the constant keys match its values with these tests. Seems like a science experiment with `arrary_flip` being kept because... That said why not feed the data providers into the tests for the factory methods?